src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -345,19 +345,35 @@
 ;;;; regarding the cause of the shutdown, and is intended to be passed back
 ;;;; in to the top-level functions that perform various shutdown steps.
 
+(def ^{:doc "A process exit request like
+  {:status 7
+   :messages [[\"something for stderr\n\" *err*]]
+              [\"something for stdout\n\" *out*]]
+              [\"something else for stderr\n\" *err*]]"}
```

Comment:
From what I've seen, it looks like most functions have doc strings below their declaration, as in `request-shutdown*` below. Why the `{:doc }` here?

Reply:
Oh, right, thanks.  I'd just forgotten that `def` also supports a docstring after the name (I believe it didn't in (much?) older clojure versions).  I'll fix it.